### PR TITLE
JBPM-9383: Mark unstable tests to be skipped by PR check

### PR DESCRIPTION
**Pull request by sutaakar**
**Thank you for submitting this pull request**

**JIRA**: 

[JBPM-9383](https://issues.redhat.com/browse/JBPM-9383)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by lucamolteni** [comment]
Out of curiosity, what is the difference between this annotation and just delete the test?

If we don't care if they're red or green to merge them then we can as well delete the tests

**Comment by sutaakar** [comment]
The difference is a downstream check for certifications. In downstream the tests pass (I guess there must be some Jenkins difference though I wasn't able to identify it).

**Comment by lucamolteni** [comment]
We could move this test to the downstream suite then.
I think we already have several test suites, such as turtle tests if I remember correctly

**Review by gmunozfe — APPROVED** [review]
Looks good, thanks @sutaakar

**Comment by sutaakar** [comment]
@lucamolteni I prefer not to split the Kie server tests, rather keep them in one place. It is easier to maintain.

**Review by mareknovotny — APPROVED** [review]

**Comment by mareknovotny** [comment]
failed tests are related to https://github.com/kiegroup/drools/pull/3133 and DROOLS-5685

@tarilabs @mariofusco have you checked the tests here in this repository? As the PR https://github.com/kiegroup/drools/pull/3133 didn't run FDB i am not sure if this is now really broken

**Comment by tarilabs** [comment]
the test failures clearly exhibited a problem which has nothing to do with kie-server, the use-case was not tested on the drools.git repo which is now done with https://github.com/kiegroup/drools/pull/3133

I will check this one locally once again (sec)

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java [hunk 1]
```diff
@@ -183,6 +185,7 @@ public void testContainerCreatedAfterStartup() throws Exception {
     }
 
     @Test
+    @Category({UnstableOnJenkinsPrBuilder.class})
```
Comment (tarilabs) [inline kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java:188]:
```suggestion
```

for me https://github.com/kiegroup/drools/commit/1730221b533e4b8da9c02f5cb8730c2490a39e7e
resolved the issue

as I taken the steps to reproduce indicated in https://issues.redhat.com/browse/DROOLS-5685

that is:
image


I get:

image

**Review by tarilabs — CHANGES_REQUESTED** [review]

**Comment by mareknovotny** [comment]
Jenkins retest this please

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java (hunk 1 above)
Comment (mareknovotny) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java:188]: ok, i will try to trigger tests again, thanks @tarilabs for checking

**Comment by mareknovotny** [comment]
Github action build has a deadlock on https://github.com/kiegroup/droolsjbpm-integration/blob/master/kie-spring-boot/kie-spring-boot-autoconfiguration/kie-server-spring-boot-autoconfiguration-audit-replication/src/test/java/org/kie/server/spring/boot/autoconfiguration/audit/replication/AuditDataReplicationKieServerTest.java reported as https://issues.redhat.com/browse/JBPM-9402

**Comment by sutaakar** [comment]
Jenkins retest this please

**Comment by tarilabs** [comment]
it seems master is failing for:
```
org.kie.integration.testcoverage.instrumentation.InstrumentationTest.testMetaInfoExists
```

although "interesting" since the only related modification was https://github.com/kiegroup/droolsjbpm-integration/pull/2247 which was green as still can be seen in the PR page?

**Comment by tarilabs** [comment]
Further data.

As mentioned previously https://github.com/kiegroup/droolsjbpm-integration/pull/2256#issuecomment-703224491

I can confirm `master` fails for:
```
org.kie.integration.testcoverage.instrumentation.InstrumentationTest.testMetaInfoExists
```

however I have good reasons to believe the failure is actually caused by: https://github.com/kiegroup/drools/pull/3090

Since it's also shown in the failure log of this PR on Jenkins: https://rhba-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/KIE/job/master/job/pullrequest/job/droolsjbpm-integration-master.pr/4/

so while this latter PR might have fixed the Karaf infrastructure, it is showing the test under discussion was failing at that time.

I am attempting a fix taking KOGITO-2853 into consideration.

**Comment by tarilabs** [comment]
verified locally and 

```
org.kie.integration.testcoverage.instrumentation.InstrumentationTest.testMetaInfoExists
```

can be fixed with: 

- https://github.com/kiegroup/drools/pull/3138
- https://github.com/kiegroup/droolsjbpm-integration/pull/2260

**Comment by sutaakar** [comment]
Jenkins retest this please

**Comment by mareknovotny** [comment]
I merged https://github.com/kiegroup/droolsjbpm-integration/pull/2261 where GA CI again works as deadlock were fixed

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java (hunk 1 above)
Comment (sutaakar) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java:188]: @tarilabs This test is not related to the issue which you reference in your comment. Is there any specific reason for removing the unstable flag here?

Comment (tarilabs) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java:188]: Sorry @sutaakar it indeed it appears I've been marking over something unrelated. Sorry for noise.

**Review by tarilabs — APPROVED** [review]

File: kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java (hunk 1 above)
Comment (tarilabs) [reply kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/KieControllerStartupIntegrationTest.java:188]: (please consider this resolved, as wrong comment 🙏 )

**Comment by sutaakar** [comment]
@mareknovotny Can you please merge?

**Comment by mareknovotny** [comment]
@sutaakar yes, i am waiting only on GA CI to finish ;)

**Pull request merged**